doc/sphinxext/missing_references.py

```diff
@@ -0,0 +1,69 @@
+"""
+This is a sphinx extension to freeze your broken reference problems
+when using ``nitpicky = True``.
+
+The basic operation is:
+
+1. Add this extension to your ``conf.py`` extensions.
+2. Add ``missing_references_freeze = True`` to your ``conf.py``
+3. Run sphinx-build. It will generate ``missing-references.json``
+    next to your ``conf.py``.
+4. Remove ``missing_references_freeze = True`` from your
+    ``conf.py`` (or set it to ``False``)
```

Comment:
I'm not really following why its "missing_references_freeze = False" when you want to use the son file and "missing_references_freeze = True" when you want to write it.  Seems the name of the var should be "write_missing_references_json" or something.

Reply:
I went with `missing_references_write_json` – sphinx recommends that configuration values be prefixed with the extension name to reduce the likelihood of conflicts between extensions.